corehq/apps/app_manager/fields.py

```diff
@@ -477,19 +487,19 @@ def _get_cases_for_apps(self, apps_by_type, as_dict=True):
                         if as_dict:
                             case_types = map(lambda c: c._asdict(), case_types)
                     case_types_by_app[app_choice.id] = case_types
-                else:
-                    all_case_types = CaseAccessors(self.domain).get_case_types()
-                    unknown_case_types = all_case_types.difference(used_case_types)
-                    unknown_case_types = map(lambda c: RMIDataChoice(
-                        id=c,
-                        text=c,
-                        data={
-                            'unknown': True,
-                        }
-                    ), unknown_case_types)
-                    if as_dict:
-                        unknown_case_types = map(lambda c: c._asdict(), unknown_case_types)
-                    case_types_by_app[self.UNKNOWN_SOURCE] = unknown_case_types
+
+        all_case_types = CaseAccessors(self.domain).get_case_types()
```

Comment:
as far as I can tell there was no reason for this to be in the loop and even resulted in sometimes wrong behavior such as adding too many case types to the "unknown"